keep-ui/app/(keep)/incidents/[id]/activity/ui/IncidentCommentInput.tsx

```diff
@@ -0,0 +1,261 @@
+"use client";
+
+import React, { useState, useCallback, useMemo, useEffect } from "react";
+import dynamic from "next/dynamic";
+import { User } from "@/app/(keep)/settings/models";
+
+// Declare Quill on the window object since quill-mention expects it to be there
+declare global {
+  interface Window {
+    Quill: any;
+  }
+}
+
+// Import required styles
+import "react-quill-new/dist/quill.snow.css";
+import "./IncidentCommentInput.scss";
+
+// Simple dynamic import for ReactQuill - we'll handle the mention module in useEffect
+const ReactQuill = dynamic(() => import("react-quill-new"), {
+  ssr: false,
+  loading: () => <div className="h-24 w-full rounded-md border border-gray-300 bg-gray-50 animate-pulse"></div>
+});
+
+/**
+ * Props for the IncidentCommentInput component
+ */
+interface IncidentCommentInputProps {
+  value: string;
+  onValueChange: (value: string) => void;
+  users: User[];
+  placeholder?: string;
+  className?: string;
+}
+
+/**
+ * A comment input component with user mention functionality
+ */
+export function IncidentCommentInput({
+  value,
+  onValueChange,
+  users,
+  placeholder = "Add a comment...",
+  className = "",
+}: IncidentCommentInputProps) {
+  const [mentionModuleReady, setMentionModuleReady] = useState(false);
+
+  useEffect(() => {
```

Comment:
@skynetigor  , I had to resort to some script tag voodoo to get quill-mention working. Regular imports were just throwing "Cannot resolve module" errors no matter what I tried. What worked is.. Manually adding the script onto the DOM, setting window.Quill first and then the loading order.

Reply:
Resolved by @Kiryous  in #4841